api/routes/auth.js

```diff
@@ -0,0 +1,149 @@
+  
+});
+
+/**
+ * 
+ * - [ ] **Create a `POST /api/login` route:** Create a new route that allows someone to login. On successful creation, return a JWT token. You should return an error in the following cases:
+  * Username is not found
+  * Username and password do not match
+ * 
+ */
+
+router.post('/login', async (req, res, next) => {
+    const status = 201;
+    try {
+        const {username, password} = req.body;
+        const user = await User.findOne({username});
+        if (!user) {
+            const error = new Error(`Username could not be found.`);
+            error.message = 404;
+            return next(error);
+        }
+        const isValid = await bcrypt.compare(password, user.password);
+        if (!isValid) {
+            const error = new Error(`Password is invalid.`);
+            error.message = 400;
+            return next(error);
+        }
+        const payload = { id: user._id }
+        const options = { expiresIn: '1 day' }
+        const token = jsonwebtoken.sign(payload, SECRET_KEY, options)
+        res.status(201).json({ status, token});
+    } catch (e) {
+        console.error(e);
+        const error = new Error(`Please check your username and password and try again.`)
+        error.status = 400;
+        next(error);
+    
+    }
+});
+
+/**
+ * 
+ * - [ ] **Create a `PATCH /api/users/:id/permissions` route:** Create a new route that allows for an admin to change permissions of another user. 
+ * The route should only be looking for the `admin: <boolean>` key in the request body and setting the value appropriately. On success, return a status 204. 
+ * You should return an error in the following cases:
+  * A valid JWT token is not provided (status 401)
+  * The JWT token is for a user who is not an admin (status 401)
+  * User cannot be found (status 404)
+  * The request body does not include an `admin` key with a boolean value (status 400)
+
+ * 
+ */
+
+router.patch('/users/:id/permissions', async (req, res, next) => {
+    const { id } = req.params;
+    try {
+        const token = req.headers.authorization.split(`Bearer `)[1];
+        const isValid = jsonwebtoken.verify(token, SECRET_KEY);
+        if (!isValid) {
+            throw new Error(`Token is not valid. Please check your request and try again.`);
+            //error.message = 401;   I continously ran into failed requests until i commented this out and changed the above from `const error = new Error...` to the current statement. Any suggestion as to why this is? This prevents me from outputing the specific error codes as intended.
+            //return next(error);
+        }
+        const { admin } = req.body;
+        if (admin == null) {
+            throw new Error(`Unable to authenticate as Admin. Please check your request and try again. `);
+            //error.message = 401;
+            //return next(error);
+        }
+        const isAdmin = await User.findOne({_id: isValid.id});
+        if (isAdmin.admin == false) {

```

Comment:
Because you're storing `admin` as a string, I imagine this is the reason for the issue.
```js
'false' == false // returns false
```
I know. JavaScript.